src/algorithm/contains.rs

```diff
@@ -0,0 +1,318 @@
+
+use types::{COORD_PRECISION, Point, LineString, Polygon, MultiPolygon};
+use algorithm::intersects::Intersects;
+
+/// Checks if the geometry A is completely inside the B geometry.
+
+pub trait Contains<RHS = Self> {
+    /// Checks if the geometry A is completely inside the B geometry.
+    ///
+    /// ```
+    /// use geo::{Coordinate, Point, LineString, Polygon};
+    /// use geo::algorithm::contains::Contains;
+    ///
+    /// let p = |x, y| Point(Coordinate { x: x, y: y });
+    /// let v = Vec::new();
+    /// let linestring = LineString(vec![p(0., 0.), p(2., 0.), p(2., 2.), p(0., 2.), p(0., 0.)]);
+    /// let poly = Polygon(linestring.clone(), v);
+    ///
+    /// //Point in Point
+    /// assert!(p(2., 0.).contains(&p(2., 0.)));
+    ///
+    /// //Point in Linstring
+    /// assert!(linestring.contains(&p(2., 0.)));
+    ///
+    /// //Point in Polygon
+    /// assert!(poly.contains(&p(1., 1.)));
+    ///
+    /// ```
+    ///
+    fn contains(&self, rhs: &RHS) -> bool;
```

Comment:
maybe we can also add a `within` method that is default implemented with the `contains` ?